test/unit/StreamrClient.test.js

```diff
@@ -118,11 +118,14 @@ describe('StreamrClient', () => {
         })
 
         c.send = (msgToSend) => {
-            const next = c.expectedMessagesToSend.shift()
-            if (!next) {
-                throw new Error(`Sending unexpected message: ${JSON.stringify(msgToSend)}`)
-            }
-            next.verificationFunction(msgToSend, next.msgToExpect)
+            return new Promise((resolve, reject) => {
+                const next = c.expectedMessagesToSend.shift()
+                if (!next) {
+                    throw new Error(`Sending unexpected message: ${JSON.stringify(msgToSend)}`)
```

Comment:
Note this doesn't cause the promise to reject, the promise hasn't even been created yet. This causes `c.send` to throw synchronously.

Might consider just changing it to an async function i.e. `c.send = async (msgToSend) => ` this will return a promise and any errors will always be converted to promise rejections